File: Data Structures & Algorithms/depth-of-binary-tree/submission-0.js
/**
 * Definition for a binary tree node.
 * class TreeNode {
 *     constructor(val = 0, left = null, right = null) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */

class Solution {
    /**
     * @param {TreeNode} root
     * @return {number}
     */
    maxDepth(root) {

        if (root === null) return 0

       const  maximunDepth= (currentRoot) => {
            if (currentRoot === null) return 0
            let maxLeft = maximunDepth(currentRoot.left)
            let maxRight = maximunDepth(currentRoot.right)
            return 1 + Math.max(maxLeft, maxRight)
        }

        return maximunDepth(root)
    }
}

File: Data Structures & Algorithms/rotate-array/submission-0.go
func rotate(nums []int, k int) {
	k=k%len(nums)
	if len(nums)==k {
		return
	}
	left,right:=0,len(nums)-1

	for left<right {
		nums[left],nums[right]=nums[right],nums[left]
		right-=1
		left+=1
	}
	left,right=0,k-1
	for left<right {
		nums[left],nums[right]=nums[right],nums[left]
		right-=1
		left+=1
	}
	left,right=k,len(nums)-1
	for left<right {
		nums[left],nums[right]=nums[right],nums[left]
		right-=1
		left+=1
	}
}
